yax/modules/Filter/Filter.py

```diff
@@ -0,0 +1,125 @@
+'''
+Created on Feb 24, 2016
+
+@author: Andrew Hodel
+'''
+import yax/TaxIDBranch as branch
+
+
+class Filter():
+    filter_results = None
+    maximum_lca_threshold = None
+    taxid_tree = None
+    read_taxids = {}
+    taxid_ancestor = {}
+    taxid_mismatch = []
+    minumum_informed_threshold = None
+    informed_read_taxids = {}
+    read = None
+
+    def main(working_dir, output):
+        # puts the taxtree in output_location
+        branch.preprocessing(nodes_fp, names_fp, gi_taxid_nucl_fp,
+                             output_location, project_name)
+        branch.get_input_taxid_data(output_location)
+        pass
+
+    def identify_informative_reads(self, survey_results,
+                                   maximum_lca_threshold):
+
+        '''
+        for nth element in the survey_results list of read fragments
+           locate in the taxid tree where that read frag hits
+           traverse up the taxid tree LCA times
+               storing each taxid that exists at that location
+                   if that taxid already has been hit then incrment a counter
+                   for the node
+        '''
+        pass
+
+    def add_informed(self, taxid):
+
+        pass
+
+    def rank_informed_reads(self,
+                            informed_reads,
+                            minimum_informed_threshold):
+        # 2 thresholds are allowed
+        # minimum overall frequency supplied by user
+        #  OR
+        # minimum relative frequency with respect to all informed reads
+        # contains hits by informed
+            # (taxid is hit by at least 1 informed read)
+        # exclusive hits by informed(taxid is hit by only informed reads)
+        # total number of hits on a taxid
+        # TODO
+
+        return self.filter_results
+
+    # TEST
+    # kind of done just need concrete inputs to complete parsing
+    # returns a list of ancestors for a given taxid and number of ancestors
+
+    def get_ancestors(self, taxids):
+        # use taxid clipper to get the tree file
+        taxid_ancestors = [[]]
+        number = 0
+        for tax in taxids:
+            # look up the ancestors for the given taxid
+            taxid_ancestors[number] = branch.get_ancestors()
```

Comment:
Ancestors are all in one place called parents in the record. You dont need to recursively move backwards to find them all.